src/java/org/apache/ivy/plugins/conflict/LatestCompatibleConflictManager.java

```diff
@@ -85,14 +85,12 @@ public LatestCompatibleConflictManager(String name, LatestStrategy strategy) {
         if (versionMatcher.isDynamic(mrid)) {
             while (iter.hasNext()) {
                 IvyNode other = iter.next();
-                if (versionMatcher.isDynamic(other.getResolvedId())) {
+                if (versionMatcher.isDynamic(other.getResolvedId())
+                        || !versionMatcher.accept(mrid, other.getResolvedId())
+                        && !handleIncompatibleConflict(parent, conflicts, node, other)) {
```

Comment:
This change doesn't look right. The previous logic would have returned `null` if the `versionMatcher.isDynamic` returned true. With this change, it will require an additional `&&` to be satisfied for the `!handlerIncompatibleConflict...`